sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -186,7 +187,12 @@ private[hive] class HiveClientImpl(
   /** Returns the configuration for the current session. */
   def conf: HiveConf = state.getConf
 
-  private val userName = conf.getUser
+  private val userName: String = try {
+    val ugi = HiveUtils.getUGI
+    ugi.getShortUserName
+  } catch {
+    case e: LoginException => throw new IOException(e)
+  }
```

Comment:
Given this is a fix for forth and back (https://github.com/apache/spark/pull/20034 and https://github.com/apache/spark/pull/17363), can you test if the metastore is MySQL as well?

Reply:
> Given this is a fix for forth and back (#20034 and #17363), can you test if the metastore is MySQL as well?

it works correctly.

Reply:
And i think it's not a fix for forth and back, actually, it is a copy of #20034  use ugi.shortUserName instead of userName